### DISCUSSION TITLE
Genmon / Longest run time without reboot

### DISCUSSION THREAD

**Discussion by Jkh62** [jgyates/genmon#1030] [Questions] [answered]
I have several routers (Linux) that run like a champ, rock solid without any reboot.
For those that have been running Genmon for a while does it play nice on the pi and does the pi run solid or does it sometimes benefit from a restart?

My other question is hitting the update tab on the about page that updates Genmon software. 
Does this update restart the pi? 

I am using a pi ZERO W. 
Since I know nothing about the pi updates to their OS does it require checking on system updates on a regular basis??

**Comment by jgyates** [comment, marked as answer]
Updating the software on the about page only updates genmon. It does not update the system software or restart the system.

In my view, based on my experience and the feedback from this forum, genmon runs without the need for periodic reboots in the vast majority to cases (hundreds). I have seen a small number (less than 10) of folks who did need a reboot, however I don't have any data on why they needed a reboot.

You are not required to update your OS. Updating linux periodically will give you bug fixes and security updates. If you do not update periodically the larger risk, in my view, is that if when there is a critical update that does need to be applied, the changes of something breaking is larger if you have not updated in a while. Most open source software is tested with only with upgrades from the previous versions. Having said that, there is nothing wrong with leaving the linux running without applying updates as long as it is not exposed to the internet and protected with a firewall / router.

The raspberry pi OS is based on Debian, which is one of the more popular linux distributions.

**Reply by Jkh62** [reply]
Thx for the reply and it's exactly what I wanted to hear. 
I might do the mod to cut down writes to the log file / SD card but everything else looks pretty solid and well constructed.

**Comment by skipfire** [comment]
@Jkh62 if you do the log2ram update you may have to change the defaults.  The directory may be above the default sizes specified which will cause log2ram to fail to start, but the pi will operate with it still as the physical locations.

**Comment by Jkh62** [comment]
I am playing catch up on how the system operates and trying to catch up on options available.
Great info.
